### PR TITLE
improvement(components): [scrollbar] native mode not observe DOM changes

### PR DESCRIPTION
**Pull request by gaoshanshanshan**
Please make sure these boxes are checked before submitting your PR, thank you!

- [] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

el-scrollbar组件在使用原生滚动条状态下，不观测dom变化重新计算滚动条高度

### CODE REVIEW

File: packages/components/scrollbar/src/scrollbar.vue [hunk 1]
```diff
@@ -140,19 +140,15 @@ export default defineComponent({
...
-    )
+    watchEffect(() => {
+      if (!props.noresize && !props.native) {
+        ;({ stop: stopResizeObserver } = useResizeObserver(resize$, update))
+        stopResizeListener = useEventListener('resize', update)
+      } else {
+        stopResizeObserver?.()
+        stopResizeListener?.()
+      }
+    })
```
Comment (JeremyWuuuuu) [inline packages/components/scrollbar/src/scrollbar.vue:151]: **Praise:** the code seems legit to me, would you mind adding a test for your code? If you don't know how to do that, I can assist you with it.

Comment (gaoshanshanshan) [reply packages/components/scrollbar/src/scrollbar.vue:151]: 😀 ，I know little about testing. I would appreciate it if you could help me.

**Review by jw-foss — COMMENTED** [review]
So the main idea is that we can assert whether the function has been called in this case we can assert something like this

Note these are all pseudo code.

```ts
it("should work", async () => {
jest.mock("@vueuse/core", () => {
  return { useEventListener: jest.fn(), useResizeObserver: jest.fn() }
}))

Setup()

expect(vueuse.useEventListener).hasBeenCalled())
})
```

**Comment by gaoshanshanshan** [comment]
@JeremyWuuuuu   I have added test cases

**Review by msidolphin — APPROVED** [review]

**Comment by yunfeng666** [comment]
everything is good

**Comment by xiaoxian521** [comment]
Hey, thank you very much for your contribution, but here are some conflicts that need to be solved by you. We can continue to review the code after solving. One thing to remind is that if the conflicts is still not resolved after `10` days, sorry, we will close this pr ❤️

**Pull request closed without merging**